### PR TITLE
Testing/6.10

### PR DESCRIPTION
**Pull request by uilianries**
Hi!

I just added support to Linux and Mac, but also build on Appveyor with gcc.

related to the : https://github.com/bincrafters/community/issues/271

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -11,26 +11,34 @@ class RagelConan(ConanFile):
                   "Ragel targets C, C++ and ASM. "
     homepage = "http://www.colm.net/open-source/ragel"
 
-    settings = {"os": ["Windows"],
+    settings = {"os": ["Windows", "Linux", "Macos"],
```
Comment (SSE4) [inline conanfile.py]: if it is an installer package, may be just os_build and arch_build are enough? as well as just one CI configuration per OS.

File: conanfile.py [hunk 2]
```diff
@@ -134,30 +152,39 @@ def build(self):
...
         build_dir = os.path.join('out','bin')
 
         self.copy(pattern="COPYING", dst="licenses", src=build_src_dir)
         self.copy(pattern="CREDITS", dst="licenses", src=build_src_dir)
-        self.copy(pattern="*.exe", dst="bin", src=build_dir, keep_path=False)
+        if self.settings.os != "Windows":
+            autotools = self.configure_autotools()
+            autotools.make(args=["install"])
+        else:
+            self.copy(pattern="*.exe", dst="bin", src=build_dir, keep_path=False)
```
Comment (SSE4) [inline conanfile.py]: doesn't `make install` work on Windows?

File: build.py [hunk 3]
```diff
@@ -1,92 +1,11 @@
+
```
Comment (SSE4) [inline build.py]: empty line, shebang has to go the first

File: conanfile.py [hunk 4]
```diff
@@ -70,10 +88,10 @@ def build(self):
             vccmd = tools.vcvars_command(self.settings) + " &&"
             compiler_cc = '$PWD/compile cl -nologo'
             compiler_cxx = compiler_cc
-            
+
         elif self.settings.compiler == 'gcc':
             vccmd = ''
-            if 'MINGW_HOME' in os.environ: 
+            if 'MINGW_HOME' in os.environ:
                 compiler_cc = tools.unix_path(self.deps_env_info["mingw_installer"].CC)
```
Comment (SSE4) [inline conanfile.py]: probably this is unnecessary, as mingw_installer already sets CC/CXX

**Review by SSE4 — CHANGES_REQUESTED** [review]

File: conanfile.py (hunk 1 above)
Comment (uilianries) [reply conanfile.py]: Good point, I should checked other installers before to do this.

File: conanfile.py (hunk 2 above)
Comment (uilianries) [reply conanfile.py]: Good question. All Windows stuff I just moved to another method. But I'll check your suggestion.

File: conanfile.py (hunk 4 above)
Comment (uilianries) [reply conanfile.py]: Good point.

**Comment by uilianries** [comment]
@SSE4 Thanks for review this PR, you really contributed for a better package. I'll make all changes at evening (10h from now).

**Comment by solvingj** [comment]
You guys are way ahead of me on this one.  It looks good.

**Comment by Croydon** [comment]
Friendly Ping 😄

File: conanfile.py (hunk 1 above)
Comment (SSE4) [reply conanfile.py]: I've changed to os_build/arch_build

File: conanfile.py (hunk 2 above)
Comment (SSE4) [reply conanfile.py]: I've changed recipe to use install everywhere

File: build.py (hunk 3 above)
Comment (SSE4) [reply build.py]: removed

File: conanfile.py (hunk 4 above)
Comment (SSE4) [reply conanfile.py]: I've removed unnecessary handling of CC

**Review by SSE4 — APPROVED** [review]
LGTM

**Pull request merged**